agent/src/main/java/com/cloud/agent/Agent.java

```diff
@@ -419,6 +419,15 @@ protected void cancelTasks() {
         }
     }
 
+    protected void cleanupAgentZoneProperties() {
+        // Unset zone, cluster and pod values so that host is not added back
+        // when service is restarted. This will be set to proper values
+        // when host is added back
```

Comment:
These commented lines could be a nice Javadoc for the method; what do you think of moving them to a Javadoc block?